### PR TITLE
ast(cli): Recursively parse AST

### PR DESCRIPTION
**Pull request by ev1stensberg**
**What kind of change does this PR introduce?**

Refactoring
**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
Yes
**Summary**
Refactors AST parsing, cutting a lot of dead code and simplifies our repository. It removes alot of tech burden and potential bug reports.

It's fun

**Does this PR introduce a breaking change?**
Probably a major

Logic still intact
**Other information**
Good stuff

### CODE REVIEW

File: lib/generate-plugin/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const yeoman = require("yeoman-environment");
-const PluginGenerator = require("../generators/plugin-generator").PluginGenerator;
+const PluginGenerator = require("../generators/plugin-generator")
```
Comment (ematipico) [inline lib/generate-plugin/index.js]: This can be destructured :)
```js
const { PluginGenerator } = require("../generators/plugin-generator")
```

Comment (ev1stensberg) [reply lib/generate-plugin/index.js]: Im not done

Comment (ematipico) [reply lib/generate-plugin/index.js]: yeah and in the meantime we give feedback while you develop :)

Comment (ev1stensberg) [reply lib/generate-plugin/index.js]: Gotcha, just don't want you to do extra work 🙌🤙

**Comment by bitpshr** [comment]
@ev1stensberg nice work so far, let us know when it’s ready to review.

File: .yo-rc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (ev1stensberg) [inline .yo-rc.json]: Ups, forgot this!

**Pull request merged**